tests/cmake_integration/Messenger/Args.h

```diff
@@ -1,98 +0,0 @@
-/*
```

Comment:
I recall attempting to remove the duplication in the CMake messenger tests. However, it was causing link issues to share the same code directory. I think that it was something to do with the export header being generated by Messenger_1 and consequently being included in Messenger_2 (thereby defining export macros when it should not). Feel free to test this theory further but I recommend reverting these changes and for now living with the duplication.

There is seemingly related link errors showing up on the [scoreboard](https://dev.azure.com/opendds/OpenDDS/_build/results?buildId=734&view=logs&jobId=567bf023-8cc0-5d40-5ba2-7f319774a15e&taskId=bdfa9db1-a1d0-5626-30bf-6f3cf19b26a7&lineStart=80&lineEnd=81&colStart=1&colEnd=1). 

It appears as though scoreboard giving a false-positive for the CMake tests—the link failures are not being flagged!

Reply:
> Feel free to test this theory further but I recommend reverting these changes and for now living with the duplication.

I just tested the branch locally on Windows and my theory is wrong! So, not sure what the problem is—but you are probably good to go on your changes.

Reply:
> I just tested the branch locally on Windows and my theory is wrong! So, not sure what the problem is—but you are probably good to go on your changes.

Sorry, I mistakenly built this using the master branch. I'm also seeing the same issues locally that the build server is reporting.

Reply:
Alright I found the issue—it's not actually the Messenger_1/2 projects which collide as I thought, but the problem is when `$DDS_ROOT/tests/DCPS/Messenger` is built in the standard MPC build, the generated artifacts (TypeSupport, etc) are clashing with the CMake-Build-Generated artifacts.

So, I think we're out of luck on this one for the moment. There may be a way to fix it, but it would require ensuring that the first directory searched for includes is the ones generated in (for example) `%DDS_ROOT%\tests\cmake_integration\Messenger\Messenger_1\build` prior to `%DDS_ROOT%\tests\DCPS\Messenger`—but it doesn't seem practical to try and enforce that.

Reply:
How about we have cmake copy everything into `cmake_integration/Messenger_X`? Also for Perl dependency we can make a script that runs perl script that runs messenger without PerlDDS.

Reply:
Ah! Good idea.